recipes/protobuf/all/conanfile.py

```diff
@@ -178,6 +231,16 @@ def _patch_sources(self):
                 "#elif PROTOBUF_GNUC_MIN(12, 0)",
                 "#elif PROTOBUF_GNUC_MIN(12, 2)")
 
+        # Add $ORIGIN/../{relative_path} in RPATH for running executables.
+        if (self.settings.os == "Linux" or is_apple_os(self)) and self.options.shared:
+            old_str = ""
+            if is_apple_os(self):
+                old_str = 'PROPERTY INSTALL_RPATH "@loader_path/../lib"'
+            elif self.settings.os == "Linux":
+                old_str = 'PROPERTY INSTALL_RPATH "$ORIGIN/../${CMAKE_INSTALL_LIBDIR}"'
+            new_str = "PROPERTY INSTALL_RPATH " + self._get_rpath()
+            replace_in_file(self, os.path.join(self.source_folder, "cmake", "install.cmake"), old_str, new_str)
```

Comment:
Last thing, I'm curious about all of the RPATH workarounds, why are they necessary? Do these changes improve the situation? I would hope that a VirtualRunEnv/VirtualBuildEnv would allow them to work unmodified, is that not the case?

Reply:
I agree that first part is a pain point, but my main concern with this change now that I've looked into how it works is that I don't know if the packages will be relocatable. Mainly since you use the relative path between two package folders. Maybe @uilianries has some insight?

Reply:
The LD_LIBRARY_PATH will be searched for, if dependency binaries are not found in its rpath.

Reply:
Personally I would prefer an option, but since I understand now that it would work either way, I'm not sure. I think I'll defer to a maintainer, but as-is I think it's okay for now.